lib/services/local/presence.go

```diff
@@ -1577,6 +1580,38 @@ func (s *PresenceService) listResources(ctx context.Context, req proto.ListResou
 	}, nil
 }
 
+// UpsertDatabaseService creates or updates (by name) a DatabaseService resource.
+func (s *PresenceService) UpsertDatabaseService(ctx context.Context, service types.DatabaseService) (*types.KeepAlive, error) {
+	if err := service.CheckAndSetDefaults(); err != nil {
+		return nil, trace.Wrap(err)
+	}
+	value, err := services.MarshalDatabaseService(service)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	item := backend.Item{
+		Key:     backend.Key(databaseServicePrefix, service.GetName()),
+		Value:   value,
+		Expires: service.Expiry(),
+		ID:      service.GetResourceID(),
+	}
+	lease, err := s.Put(ctx, item)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	if service.Expiry().IsZero() {
+		return &types.KeepAlive{}, nil
+	}
+	return &types.KeepAlive{
+		Type:      types.KeepAlive_DATABASE,
```

Comment:
Right now KeepAlive_DATABASE Will be reused between 
` database servers` and `database services` so we should take extra measure and ensure that keys we won't use the same keys for database services and servers.

Reply:
Their prefix is different, so it's unlikely that we mix them.

It crossed my mind to create a new KeepAlive type
Do you think it's better?

Reply:
We we can ensure that we don't mix them I think that it is fine to reuse the `KeepAlive_DATABASE`